lib/travis/build/script/dart.rb

```diff
@@ -88,61 +150,111 @@ def announce
               sh.cmd 'sh -e /etc/init.d/xvfb start'
               # give xvfb some time to start
               sh.cmd 't=0; until (xdpyinfo -display :99 &> /dev/null || test $t -gt 10); do sleep 1; let t=$t+1; done'
               sh.cmd 'pub run test -p vm -p content-shell -p firefox'
             else
-              sh.cmd 'pub run test'
+              pub_run_test
             end
           end
           # tests with test_runner for old tests written with unittest package
-          sh.elif '[[ -d packages/unittest ]] || grep -q ^unittest: .packages 2> /dev/null', raw: true do
+          sh.elif package_installed?('unittest'), raw: true do
+            sh.fold "deprecated.unittest" do
+              sh.deprecate <<MESSAGE
+DEPRECATED: The unittest package is deprecated. Please upgrade to the test
+package. See https://github.com/dart-lang/test#readme.
+MESSAGE
+            end
+
             sh.fold 'test_runner_install' do
               sh.echo 'Installing Test Runner', ansi: :yellow
               sh.cmd 'pub global activate test_runner'
             end
 
-            if with_content_shell
+            if config[:with_content_shell]
               sh.cmd 'xvfb-run -s "-screen 0 1024x768x24" pub global run test_runner --disable-ansi'
             else
               sh.cmd 'pub global run test_runner --disable-ansi --skip-browser-tests'
             end
           end
+
+          dartanalyzer
+          dartfmt
         end
 
         private
 
+          def pub_run_test
+            return unless (args = task[:test])
+            args = args.is_a?(String) ? " #{args}" : ""
+
+            # Mac OS doesn't need or support xvfb-run.
+            xvfb_run = 'xvfb-run -s "-screen 0 1024x768x24" '
+            xvfb_run = '' if task[:xvfb] == false || os == "macos"
+            sh.cmd "#{xvfb_run}pub run test#{args}"
+          end
+
+          def dartanalyzer
+            return unless (args = task[:dartanalyzer])
+            args = '.' unless args.is_a?(String)
+            sh.cmd "dartanalyzer #{args}"
+          end
+
+          def dartfmt
+            return unless (args = task[:dartfmt])
+            if args.is_a?(String)
+              sh.echo "dartfmt arguments aren't supported.", ansi: :red
+            end
+
+            sh.if package_installed?('dart_style'), raw: true do
+              sh.raw 'function dartfmt() { pub run dart_style:format "$@"; }'
+            end
+
+            sh.cmd 'unformatted=`dartfmt -n .`'
+            sh.if '! -z "$unformatted"' do
+              sh.echo "Files are unformatted:", ansi: :red
+              sh.echo "$unformatted"
+              sh.failure ""
+            end
+          end
+
+          def package_installed?(package)
+            "[[ -d packages/#{package} ]] || grep -q ^#{package}: .packages 2> /dev/null"
+          end
+
+          def os
+            config[:os] == 'osx' ? 'macos' : 'linux'
+          end
+
           def archive_url
             urlEnd = ''
```

Comment:
I didn't notice this until now, but can we move to snake case, please? (Did I miss any other?)